pkg/invariants/types.go

```diff
@@ -0,0 +1,43 @@
+package invariants
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+type InvariantTest interface {
```

Comment:
I think we need better naming here, Invariant collides hard with a whole bunch of other tests which are not represented as this InvariantTest or in this InvariantRegistry.

Would InClusterInvariantTest be better? With clear godoc that this is for a subset of complex invariant tests that require resources running in the cluster, and collecting data from them at the end of the suite.

Reply:
> I think we need better naming here, Invariant collides hard with a whole bunch of other tests which are not represented as this InvariantTest or in this InvariantRegistry.
> 
> Would InClusterInvariantTest be better? With clear godoc that this is for a subset of complex invariant tests that require resources running in the cluster, and collecting data from them at the end of the suite.

Nothing about this limits us to in-cluster only.  It has the option of making in-cluster stuff, but it could also do nothing.

Reply:
And given that, would you like to collapse them here?